_source/_posts/2019-12-30-java-jersey-jaxrs.adoc

```diff
@@ -0,0 +1,595 @@
+
+With the task created again, execute the following command in your terminal:
+
+[source,text]
+----
+curl -X GET http://localhost:8080/tasks
+----
+
+Your response should be an array with all the tasks you created so far. In my case, the result was:
+
+[source,json]
+----
+[{"id":"d7fc8d86-d7fe-47b9-a6ac-f5e8e28e2ea9","description":"do the dishes"}]
+----
+
+Now that you can both create and list all tasks, the next step is to update an existing task.
+
+== Update an Entry
+
+To update the task you are going to create a PUT request to the `tasks/<task_id>` URI, where `<task_id>` is the ID of the task you want to update.
+
+Inside the `TaskResource`, add the following method:
+
+====
+[source,java]
+----
+@PUT // <1>
+@Path("/{taskId}") // <2>
+public Response updateTask(@PathParam("taskId") UUID taskId, TaskRequest request) { // <3>
+    if (!tasks.containsKey(taskId)) {
+        // return 404
+        return Response.status(Response.Status.NOT_FOUND).build(); // <4>
+    }
+
+    Task task = tasks.get(taskId);
+    task.setDescription(request.getDescription());
+
+    // return 204
+    return Response.noContent().build();
+}
+----
+
+<1> As happened in the other methods, the `@PUT` annotation defines which HTTP method is going to be used in the endpoint.
+
+<2> You're also defining the `@Path` annotation, which will contain the ID of the task you want to update. Since this is a dynamic value, you want to get it inside a variable.
+
+<3> Jersey allows you to do that by defining an argument in the method and use the annotation `@PathParam` on it, informing the name of the param you want to retrieve.
+
+<4> The `updateTask` method will search for a task with the given id inside the current tasks. If it finds, it proceeds to update it with the new description, otherwise, it will return a `404 - Not Found` response to the user.
+====
+
+It's time to test if the endpoint is working. Start your application again and execute the following code:
+
+[source,text]
+----
+curl -X POST \
+  http://localhost:8080/tasks \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description" : "do the dishes"
+}'
+----
+
+The command above will create a new task, just like you did before. Now that you have created a task again, you can update its description using the following command:
+
+[source,text]
+----
+curl -X PUT \
+  http://localhost:8080/tasks/<task_id> \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description" : "clean the house"
+   }'
+----
+
+Replace `<task_id>` with the ID of one of the tasks you created previously.
+
+Great job! If you list your tasks again you'll see that the description changed.
+
+You implemented all the CRUD functions, except for the last one. Let's finish it by implementing the delete endpoint.
+
+== Delete a Entry
```

Comment:
Delete a Task might be better.

Reply:
"Entry" matches the others headers, but i did fix the "a" -> "an" issue